.github/workflows/test-n-publish.yml

```diff
@@ -73,26 +72,59 @@ jobs:
       with:
         repository: 'spine-tools/${{ matrix.pkg }}'
         path: ${{ matrix.pkg }}
-        ref: ${{ inputs['Spine-Toolbox'] }}
+        ref: ${{ inputs[matrix.pkg] }}
     - name: Set up Python
       uses: actions/setup-python@v4
       with:
-        python-version: '3.9'
-    - name: Install pytest and built wheels
+        python-version: '3.8'
+    - name: Install pytest and built wheels on *NIX
+      if: runner.os != 'Windows'
       run: |
         python -m pip install --upgrade pip
-        pip install dist/*/*.whl
-        pip install pytest
-    - name: Remove source from checkouts
-      # FIXME: What is the Windows equivalent?
+        python -m pip install dist/*/*.whl
+        python -m pip install pytest
+    - name: Install pytest and built wheels on Windows
+      if: runner.os == 'Windows'
+      run: |
+        python -m pip install --upgrade pip
+        Get-ChildItem ./dist/*.whl -Recurse | ForEach-Object {python -m pip install $_}
+        python -m pip install pytest
+    - name: Install additional packages for Linux
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y libegl1
+    - name: Remove source from checkouts for *NIX
+      if: runner.os != 'Windows'
       run: |
         rm -rf Spine-Toolbox/spinetoolbox* spine-items/spine_items \
           spine-engine/spine_engine Spine-Database-API/spinedb_api
-    - name: Test wheels
+    - name: Remove source from checkouts for Windows
+      if: runner.os == 'Windows'
+      run: |
+        Remove-Item Spine-Toolbox\spinetoolbox* -Recurse -Force -ErrorAction Ignore
+        Remove-Item spine-items\spine_items -Recurse -Force -ErrorAction Ignore
+        Remove-Item spine-engine\spine_engine -Recurse -Force -ErrorAction Ignore
+        Remove-Item Spine-Database-API\spinedb_api -Recurse -Force -ErrorAction Ignore
+    - name: Test wheels on *NIX
+      if: runner.os != 'Windows'
       # FIXME: how to run execution tests for Spine-Toolbox
+      env:
+        QT_QPA_PLATFORM: offscreen
       run: |
-        python -m unittest discover -s ${{ matrix.pkg }}
-
+        # FIXME: The cd stuff below is a workaround for bug in spinetoolbox tests
+        cd ${{ matrix.pkg }}
+        python -m unittest discover --verbose
+        cd ..
+        # FIXME: Use the line below once spine-tools/Spine-Toolbox#2274 is fixed
+        #python -m unittest discover -s ${{ matrix.pkg }} --verbose
+    - name: Test wheels on Windows
+      if: runner.os == 'Windows'
+      # FIXME: how to run execution tests for Spine-Toolbox
```

Comment:
If I recall correctly, if using `pytest`, execution tests are discovered automatically.  I'll submit a few PRs soon enabling test execution using both `unittest` & `pytest`.  We can update this after that.

Reply:
You could have a look how the execution tests are run in Toolbox CI. The difference is just that you need to add a special pattern to discover the execution tests, something like

´´´
if: matrix.pkg == 'Spine-Toolbox':
run: |
  cd Spine-Toolbox
  python -m unittest discover --pattern execution_test.py --verbose
´´´

Reply:
Yes, I'm aware of that.  I think the automatic discovery in pytest is a bit more "smart", and there's no need to provide a pattern.  Some simply `pytest` in the top-level directory works.